src/features/loading.ts

```diff
@@ -0,0 +1,402 @@
+/*
```

Comment:
Just a nit, this would have been easier to read broken out into separate PRs. But, it was broken out into separate commits, so it was pretty easy to just go back there to take a look:

- https://github.com/GoogleWebComponents/model-viewer/pull/487/commits/3ad1274c0e784cd1e957769ef0f981079f359fe9
- https://github.com/GoogleWebComponents/model-viewer/pull/487/commits/38c9e2bf9dbe32061925a90d4b9e2dfc8d8636e5#diff-5fa0cbec670bc4c1c710be11a06e2376

Reply:
Yeah, agreed. Lot's of regret.

Reply:
:)